src/test/java/org/ladocuploader/app/journeys/LaDocUploadFlowJourneyTest.java

```diff
@@ -250,7 +250,7 @@ void fullUploadDocumentFlow() {
     assertThat(testPage.getTitle()).isEqualTo("Upload documents");
     assertThat(testPage.findElementById("form-submit-button").getAttribute("class").contains("display-none")).isTrue();
     uploadJpgFile();
-    boolean isHidden = testPage.findElementsByButtonText("I'm finished uploading").getAttribute("class").contains("display-none");
+    boolean isHidden = testPage.findElementById("form-submit-button").getAttribute("class").contains("display-none");
     if (isHidden) {
```

Comment:
We discovered that if the code runs quickly and the element isn't there yet, then `isHidden` will be true. By using `findElementById` we guarantee that we get the element, which seems to work better.  

It is possible that the system has had a chance to take out the "display-none" yet... but this test passes now, so the gap is much smaller.